src/components/TagList.tsx

```diff
@@ -13,28 +14,30 @@ const TagList: (props: TagListProps) => JSX.Element = ({
   item,
 }: TagListProps) => {
   const dispatch = useDispatch();
+  const { userData } = useSelector((state: ApplicationState) => state);
```

Comment:
In here I'd try to assign already the `state.userData.user.uid`.
There are some benefits:
- we have less data in memory
- we have fewer rerenderings - the component rerenders if the property we're binding to changes, so imagine that some action changes say `state.userData.user.email`in some other part of the app, then also this component rerenders which is not necessary